examples/temperature-measurement-app/esp32/README.md

```diff
@@ -93,41 +82,102 @@ If packages are already installed then simply activate it.
-
 4.  Use
+    [python based device controller](https://github.com/project-chip/connectedhomeip/tree/master/src/controller/python)
+    or
     [standalone chip-tool](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
     or
-    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin)
+    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin/CHIPTool)
+    or
+    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
     to communicate with the device.
 
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
-    $ idf.py erase_flash ESPPORT=/dev/ttyUSB0
+    $ idf.py erase_flash ESPPORT=/dev/tty.SLAB_USBtoUART
+
+### Setting up python controller
+
+Once ESP32 is up and running, we need to set up a device controller to perform
+commissioning and cluster control.
+
+-   Set up python controller.
+
+           $ cd {path-to-connectedhomeip}
+           $ ./scripts/build_python.sh -m platform
+
+-   Execute the controller.
+
+           $ source ./out/python_env/bin/activate
+           $ chip-device-ctrl
+
+### Commissioning using BLE
+
+-   Establish the secure session over BLE. BLE is the default mode in the
+    application and is configurable through menuconfig.
+
+         - chip-device-ctrl > ble-scan
+         - chip-device-ctrl > connect -ble 3840 20202021 135246
+
+         Parameters:
+         1. Discriminator: 3840 (configurable through menuconfig)
+         2. Setup-pin-code: 20202021 (configurable through menuconfig)
+         3. Node ID: Optional.
+            If not passed in this command, then it is auto-generated by the controller and displayed in the output of connect.
+            The same value should be used in the next commands.
+            We have chosen a random node ID which is 135246.
+
+-   Add credentials of the Wi-Fi network you want the ESP32 to connect to, using
+    the `AddWiFiNetwork` command and then enable the ESP32 to connect to it
+    using `EnableWiFiNetwork` command. In this example, we have used `TESTSSID`
+    and `TESTPASSWD` as the SSID and passphrase respectively.
+
+         - chip-device-ctrl > zcl NetworkCommissioning AddWiFiNetwork 135246 0 0 ssid=str:TESTSSID credentials=str:TESTPASSWD breadcrumb=0 timeoutMs=1000
+
+         - chip-device-ctrl > zcl NetworkCommissioning EnableNetwork 135246 0 0 networkID=str:TESTSSID breadcrumb=0 timeoutMs=1000
+
+-   Close the BLE connection to ESP32, as it is not required hereafter.
+
+         - chip-device-ctrl > close-ble
+
+-   Resolve DNS-SD name and update address of the node in the device controller.
+
+         - chip-device-ctrl > resolve 0 135246
+
+### Cluster control
+
+-   The demo application supports temperaturemeasurement and basic cluster.
```

Comment:
temperature-measurement?

Reply:
Renamed it as it is present for the python controller ZCL cluster.